### PR TITLE
Update ResourceManager and Identity SDK to T2

### PR DESCRIPTION
**Pull request by v-yilinhu**
1.Update ResourceManager and Identity SDK to T2

2.Update files:

- ./DeployArmTemplate/DeployArmTemplate/DeployArmTemplate.csproj
- ./DeployArmTemplate/DeployArmTemplate/DeploymentHelper.cs
- ./DeployArmTemplate/DeployArmTemplate/packages.config

@scottaddie,@AlexGhiondea,@archerzz and @christothes for notification.

### CODE REVIEW

File: DeployArmTemplate/DeployArmTemplate/DeployArmTemplate.csproj [hunk 1]
```diff
@@ -1,122 +1,27 @@
...
-  <Target Name="EnsureBclBuildImported" BeforeTargets="BeforeBuild" Condition="'$(BclBuildImported)' == ''">
-    <Error Condition="!Exists('..\packages\Microsoft.Bcl.Build.1.0.14\tools\Microsoft.Bcl.Build.targets')" Text="This project references NuGet package(s) that are missing on this computer. Enable NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=317567." HelpKeyword="BCLBUILD2001" />
-    <Error Condition="Exists('..\packages\Microsoft.Bcl.Build.1.0.14\tools\Microsoft.Bcl.Build.targets')" Text="The build restored NuGet packages. Build the project again to include these packages in the build. For more information, see http://go.microsoft.com/fwlink/?LinkID=317568." HelpKeyword="BCLBUILD2002" />
-  </Target>
+  <ItemGroup>
+    <PackageReference Include="Azure.Identity" Version="1.5.0" />
+    <PackageReference Include="Azure.ResourceManager.Resources" Version="1.0.0-beta.5" />
+    <PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
+    <PackageReference Include="Microsoft.Rest.ClientRuntime" Version="2.3.23" />
+    <PackageReference Include="Microsoft.Rest.ClientRuntime.Azure" Version="3.3.19" />
```
Comment (archerzz) [inline DeployArmTemplate/DeployArmTemplate/DeployArmTemplate.csproj]: Please check if all those libraries are required. Thanks.

**Review by archerzz — APPROVED** [review]

**Comment by v-xuto** [comment]
@archerzz Could you help merge this PR?

**Comment by archerzz** [comment]
@v-xuto I don't have merge privilege. You may check the merge history and find the corresponding focal.

**Comment by v-xuto** [comment]
@jongio Please help merge this PR. Thanks a lot.

**Pull request merged**